library/module_utils/oneview.py

```diff
@@ -732,7 +732,8 @@ def __replace_volume_attachment_names_by_uri(self, data):
         volume_attachments = (data.get('sanStorage') or {}).get('volumeAttachments') or []
         if len(volume_attachments) > 0:
             for volume in volume_attachments:
-                self.__replace_name_by_uri(volume, 'volumeName', self.VOLUME_NOT_FOUND, self.oneview_client.volumes)
+                if volume.get('volumeUri', 'Replace'):
```

Comment:
Looks good and solves the issue, but since this will not be idempotent we might wish to warn the user that creating a volume during OV SP creation is not advised through Ansible, and makes it non-idempotent.